lb_content_resolver/subsonic.py

```diff
@@ -135,16 +135,45 @@ def update_recordings(self, recordings):
             Updates recording_id, subsonic_id, last_update
         """
 
-        recordings = [(r[0], r[1], datetime.datetime.now()) for r in recordings]
+        recording_index = { r[0]:r[1] for r in recordings }
 
         cursor = db.connection().cursor()
         with db.atomic() as transaction:
-            cursor.executemany(
-                """INSERT INTO recording_subsonic (recording_id, subsonic_id, last_updated)
-                                        VALUES (?, ?, ?)
-                     ON CONFLICT DO UPDATE SET recording_id = excluded.recording_id
-                                             , subsonic_id = excluded.subsonic_id
-                                             , last_updated = excluded.last_updated""", recordings)
+
+            placeholders = ",".join(("?", ) * len(recording_index.keys()))
+            cursor.execute("""SELECT recording_id
+                                FROM recording_subsonic
+                               WHERE recording_id in (%s)""" % placeholders, tuple(recording_index.keys()))
+            existing_ids = { row[0]:None for row in cursor.fetchall() }
+            existing_recordings = []
+            new_recordings = []
+            for r in recordings:
+                if r[0] in existing_ids:
+                    existing_recordings.append((r[0], r[1], datetime.datetime.now(), r[0]))
+                else:
+                    new_recordings.append((r[0], r[1], datetime.datetime.now()))
```

Comment:
I'm not fond of numeric indexes as they hide infos, think about using `namedtuple`.

Reply:
AH, good idea. I'll do that for the next project. this particular piece of code has a life-time of 6-12 months when sqlite is updated enough so we can get this of this entire PR and go back to the single simple query.